### PR TITLE
Replaces the Partner Auth with the new Token Auth

### PR DESCRIPTION
**Pull request by madhavc**

### CODE REVIEW

File: src/main/java/com/opentok/OpenTok.java [hunk 1]
```diff
@@ -48,9 +45,9 @@
     private String apiSecret;
     protected HttpClient client;
     static protected ObjectReader archiveReader = new ObjectMapper()
-            .reader(Archive.class);
+            .readerFor(Archive.class);
```
Comment (aoberoi) [inline src/main/java/com/opentok/OpenTok.java]: was this a deprecation or something?

File: src/main/java/com/opentok/OpenTok.java [hunk 2]
```diff
@@ -125,7 +122,7 @@ private OpenTok(int apiKey, String apiSecret, HttpClient httpClient) {
      */
     public String generateToken(String sessionId, TokenOptions tokenOptions) throws OpenTokException {
         List<String> sessionIdParts = null;
-        if (sessionId == null || sessionId == "") {
+        if(sessionId == null || sessionId == "") {
```
Comment (aoberoi) [inline src/main/java/com/opentok/OpenTok.java]: spacing

File: src/main/java/com/opentok/OpenTok.java [hunk 3]
```diff
@@ -358,12 +352,6 @@ public ArchiveList listArchives(int offset, int count) throws OpenTokException {
         String archives = this.client.getArchives(offset, count);
         try {
             return archiveListReader.readValue(archives);
-
-        // if we only wanted Java 7 and above, we could DRY this into one catch clause
-        } catch (JsonMappingException e) {
-            throw new RequestException("Exception mapping json: " + e.getMessage());
-        } catch (JsonParseException e) {
-            throw new RequestException("Exception mapping json: " + e.getMessage());
```
Comment (aoberoi) [inline src/main/java/com/opentok/OpenTok.java]: 👍

File: src/main/java/com/opentok/OpenTok.java (hunk 1 above)
Comment (madhavc) [inline src/main/java/com/opentok/OpenTok.java]: yeah this was deprecated.

File: src/main/java/com/opentok/OpenTok.java (hunk 2 above)
Comment (madhavc) [inline src/main/java/com/opentok/OpenTok.java]: good catch

Comment (madhavc) [inline src/main/java/com/opentok/OpenTok.java]: !fix

**Comment by aoberoi** [comment]
fixes #114

File: src/test/java/com/opentok/test/Helpers.java [hunk 4]
```diff
@@ -52,14 +57,32 @@ public static boolean verifyTokenSignature(String token, String apiSecret) throw
...
+
+    public static Map<String, Object> decodeToken(String token, Integer apiKey, String apiSecret)
+            throws UnsupportedEncodingException, InvalidJwtException {
+        return getClaims(token, apiKey, apiSecret);
+    }
+
+    public static boolean verifyTokenSignature(String token, Integer apiKey, String apiSecret) {
+
+        try {
+            getClaims(token, apiKey, apiSecret);
```
Comment (aoberoi) [inline src/test/java/com/opentok/test/Helpers.java]: It's a bit weird that the `getClaims()` return value is completely ignored. It's a code smell. Can we figure out how to separate pulling the claims out from verifying the token signature?

File: src/test/java/com/opentok/test/OpenTokTest.java [hunk 5]
```diff
@@ -232,15 +240,8 @@ public void testCreateBadSession() throws OpenTokException {
 
     @Test
     public void testTokenDefault() throws
-            OpenTokException, UnsupportedEncodingException, NoSuchAlgorithmException, SignatureException,
-            InvalidKeyException {
-
-        int apiKey = 123456;
```
Comment (aoberoi) [inline src/test/java/com/opentok/test/OpenTokTest.java]: please bring these variables back. this was intentional code duplication.

**Comment by aoberoi** [comment]
thanks @madhavc. as we discussed earlier, we should change the API of the Helpers such that there is one method that verifies the JWT header, and it will do the following: 1) verify the signature, 2) decode the claims, 3) verify that the claim values are correct.

File: src/main/java/com/opentok/OpenTok.java (hunk 2 above)
Comment (madhavc) [inline src/main/java/com/opentok/OpenTok.java]: !fixed

**Comment by aoberoi** [comment]
👍

**Pull request merged**